quinn/src/endpoint.rs

```diff
@@ -428,100 +441,24 @@ pub(crate) struct Shared {
-                            let mut response_buffer = BytesMut::new();
-                            match self.inner.handle(
-                                now,
-                                meta.addr,
-                                meta.dst_ip,
-                                meta.ecn.map(proto_ecn),
-                                buf,
-                                &mut response_buffer,
-                            ) {
-                                Some(DatagramEvent::NewConnection(incoming)) => {
-                                    if self.incoming.len() < MAX_INCOMING_CONNECTIONS {
-                                        self.incoming.push_back(incoming);
-                                    } else {
-                                        let transmit =
-                                            self.inner.refuse(incoming, &mut response_buffer);
-                                        self.transmit_state.respond(transmit, response_buffer);
-                                    }
-                                }
-                                Some(DatagramEvent::ConnectionEvent(handle, event)) => {
-                                    // Ignoring errors from dropped connections that haven't yet been cleaned up
-                                    let _ = self
-                                        .connections
-                                        .senders
-                                        .get_mut(&handle)
-                                        .unwrap()
-                                        .send(ConnectionEvent::Proto(event));
-                                }
-                                Some(DatagramEvent::Response(transmit)) => {
-                                    self.transmit_state.respond(transmit, response_buffer);
-                                }
-                                None => {}
-                            }
-                        }
-                    }
-                }
-                Poll::Pending => {
-                    break;
-                }
-                // Ignore ECONNRESET as it's undefined in QUIC and may be injected by an
-                // attacker
-                Poll::Ready(Err(ref e)) if e.kind() == io::ErrorKind::ConnectionReset => {
-                    continue;
-                }
-                Poll::Ready(Err(e)) => {
-                    return Err(e);
-                }
-            }
-            if !self.recv_limiter.allow_work() {
-                self.recv_limiter.finish_cycle();
-                return Ok(true);
-            }
-        }
-
-        self.recv_limiter.finish_cycle();
-        Ok(false)
-    }
-
     fn drive_send(&mut self, cx: &mut Context) -> Result<bool, io::Error> {
         self.send_limiter.start_cycle();
 
         let result = loop {
-            if self.transmit_state.outgoing.is_empty() {
+            if self.recv_state.transmit_state.outgoing.is_empty() {
```

Comment:
Feels weird for `drive_send()` to reference `recv_state`... Should it be named something else?